adapters/sspbc/sspbc.go

```diff
@@ -0,0 +1,407 @@
+package sspbc
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const version = "5.6"
+
+// MC payload (for banner ads)
+type McAd struct {
+	Id      string             `json:"id"`
+	Seat    string             `json:"seat"`
+	SeatBid []openrtb2.SeatBid `json:"seatbid"`
+}
+
+// Adslot data (oneCode detection)
+type AdSlotData struct {
+	PbSlot string `json:"pbslot"`
+	PbSize string `json:"pbsize"`
+}
+
+// Banner Template payload
+type TemplatePayload struct {
+	SiteId  string `json:"siteid"`
+	SlotId  string `json:"slotid"`
+	AdLabel string `json:"adlabel"`
+	PubId   string `json:"pubid"`
+	Page    string `json:"page"`
+	Referer string `json:"referer"`
+	McAd    string `json:"mcad"`
+}
+
+// Ext data in request.imp
+type SsbcRequestImpExt struct {
+	Data AdSlotData `json:"data"`
+}
+
+// Ext data added by proxy
+type SsbcResponseExt struct {
+	AdLabel     string `json:"adlabel"`
+	PublisherId string `json:"pubid"`
+	SiteId      string `json:"siteid"`
+	SlotId      string `json:"slotid"`
+}
+
+type SspbcAdapter struct {
+	version  string
+	endpoint string
+	// adslots mapping
+	// map key is slot id (as sent and received from proxy)
+	adSlots        map[string]AdSlotData
+	adSizes        map[string]int
+	bannerTemplate *template.Template
+}
+
+// ---------------ADAPTER INTERFACE------------------
+// Builder builds a new instance of the sspBC adapter
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	// find path to template file
+	pathToTemplate := "./adapters/sspbc/bannerTemplate.html"
+	workingDir, err := os.Getwd()
```

Comment:
Loading the file seems complicated. Please consider hardcoding as a constant (in it's own file if you prefer) and parse the template from the string. Then you don't need to worry about the file system at all or handle the difference between tests and production.

Reply:
OK, renamed.